### ISSUE TITLE
Does SSL need to be enabled when using server side encryption to put s3 object?

### ISSUE THREAD

**Issue by yunj** [localstack/localstack#2548]
```
new PutObjectRequest()
 {
      ...
      ServerSideEncryptionMethod = ServerSideEncryptionMethod.AWSKMS,
      ServerSideEncryptionKeyManagementServiceKeyId = KeyId
 };
```
When using the .NET S3 client to put the object with server side encryption, it complains the http header `x-amz-server-side-encryption-aws-kms-key-id` can only be transimitted over https. Does it mean I need to set `USE_SSL` to true and set up certificate?

Thanks.

```
level="Error" message="Failed to put object in s3 bucket" exception="Amazon.Runtime.AmazonClientException: Request contains headers which can only be transmitted over HTTPS: x-amz-server-side-encryption-aws-kms-key-id
   at Amazon.S3.Internal.AmazonS3PostMarshallHandler.ValidateSseKeyHeaders(IRequest request)
   at Amazon.S3.Internal.AmazonS3PostMarshallHandler.ProcessRequestHandlers(IExecutionContext executionContext)
   at Amazon.S3.Internal.AmazonS3PostMarshallHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CallbackHandler.<InvokeAsync>d__9`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Amazon.S3.Internal.AmazonS3ExceptionHandler.<InvokeAsync>d__1`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Amazon.Runtime.Internal.ErrorCallbackHandler.<InvokeAsync>d__5`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Amazon.Runtime.Internal.MetricsHandler.<InvokeAsync>d__1`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

**Comment by whummer** [comment]
Thanks for reporting @yunj . Looks like the .NET S3 client requires you to access the API via https only. Ideally, you should be using `USE_SSL=1`, although in newer versions of LocalStack the APIs support socket multiplexing, i.e., can be access via both - HTTPS and plain HTTP. Also, there should be no need to configure an SSL certificate, if you can instruct the .NET client to ignore self-signed certificates. Hope that helps - can you please give it a try and reopen this issue if the problem persists. Thanks

**Comment by yunj** [comment]
Thanks @whummer. Just wondering if the newer version been released or not? Currently I’m on 0.11.2.
